libs/full/collectives/include/hpx/collectives/all_to_all.hpp

```diff
@@ -42,8 +42,9 @@ namespace hpx { namespace collectives {
     hpx::future<std::vector<std::decay_t<T>>>
     all_to_all(char const* basename, T&& result,
         std::size_t num_sites = std::size_t(-1),
+        std::size_t this_site = std::size_t(-1),
         std::size_t generation = std::size_t(-1),
-        std::size_t this_site = std::size_t(-1), std::size_t root_site = 0);
+        std::size_t root_site = 0);
```

Comment:
This looks very easy to get wrong at the call site. Would it be worth considering some types for those parameters (not for this PR)? Just putting the idea out there.

Reply:
Yes, I agree. I did think of this as well. Let me introduce such types now (before the release), so we don't have to change it later on.

Reply:
That's in place now. Please re-review.